### PR TITLE
Update TypeScript dependency to 4.8

### LINKED ISSUE #810: Support TypeScript 4.8
## 🧹 Tech Debt

As per [issue 6365](https://github.com/microsoft/fast/issues/6365), @microsoft/fast-foundation is not compatible with TS 4.8.

We need to support TypeScript 4.8 to support Angular 15 where [TS 4.8 is the minimum support version](https://angular.io/guide/update-to-version-15#angular-v15-supports-typescript-version-48-or-later).

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

Resolves #810 

## 👩‍💻 Implementation

- Update dependency of `@microsoft/fast-foundation` to the new version that supports TypeScript 4.8
- Update dependency of TypesScript to 4.8

## 🧪 Testing

- Ran the nimble pipeline
- Used a locally built version of nimble to build SystemLinkShared to verify an upgrade to a new version of nimble will not immediately cause any build errors
- Used a locally built version of nimble to build SystemLinkShared with TypeScript 4.8 to verify that SystemLinkShared can update to TS 4.8 with this change

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by mollykreis** [comment]
@atmgrifter00, can you buddy this change?

**Review by atmgrifter00 — APPROVED** [review]

**Review by jattasNI — APPROVED** [review]

File: angular-workspace/package.json [hunk 1]
```diff
@@ -57,6 +57,6 @@
     "ng-packagr": "^14.1.0",
     "playwright": "^1.30.0",
     "rollup": "^3.10.1",
-    "typescript": "~4.7.4"
+    "typescript": "~4.8.0"
```
[the remark this answers is not in the document]
Comment (mollykreis) [reply angular-workspace/package.json:60]: I actually did test SystemLinkShared with & without a TypeScript upgrade along with a local build of nimble with these changes. Both built without any problems. I've updated the description to be clearer.

Comment (rajsite) [inline angular-workspace/package.json:60]: If we mark it as "~4.8.2" everywhere that'll make it so we are in a supported typescript version range for Angular >=14.2 and <16
Looks like the lock file is resolving to 4.8.4 so shouldn't need retesting, but will prevent any weirdness accidentally dropping to less than 4.8.2
https://angular.io/guide/versions

**Review by rajsite — APPROVED** [review]

**Comment by jattasNI** [comment]
@mollykreis I don't think you need to wait for @fredvisser or @TrevorKarjanis to complete this PR.

**Pull request merged**